SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp

```diff
@@ -220,7 +220,7 @@ TArray<FWorkerComponentData> EntityFactory::CreateEntityComponents(USpatialActor
 	// Actors with bNetLoadOnClient=false also need a StablyNamedObjectRef for linking in the case of loading from a snapshot or the server crashes and restarts.
 	TSchemaOption<FUnrealObjectRef> StablyNamedObjectRef;
 	TSchemaOption<bool> bNetStartup;
-	if (Actor->HasAnyFlags(RF_WasLoaded) || Actor->bNetStartup)
+	if (Actor->HasAnyFlags(RF_WasLoaded) || Actor->bNetStartup || Class->HasAnySpatialClassFlags(SPATIALCLASS_Singleton))
```

Comment:
In order for singletons to be linked in the same way as startup Actors without all the GSM plumbing, we need to make sure we set them as stably named

Reply:
How are we sure that the singleton's outer name is also stable ?

Reply:
I'm not sure we are, though I think we should maybe just have that as a requirement of using singletons in the doc, i.e. you must set the outer as something which'll live as long as a singleton, so another singleton or leave it as NULL which'll be set to the persistent level? Not too familiar with how outer objects work so would need some input from others

Reply:
Was looking at how we spawned the spatial debugger, which was done through GetWorld()->SpawnActor<>(). That would create a dynamic actor, right ?

Reply:
Yep - that'd be a dynamic actor @ImprobableNic 
Do we have a need for singletons anymore? I'm more of the opinion we should remove them entirely, and leave it up to the user.